pants-plugins/stevedore_extensions/target_types_rules.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2023 The StackStorm Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# repurposed from pants.backend.python.target_types_rules
+import dataclasses
+import os
+from dataclasses import dataclass
+
+from pants.backend.python.dependency_inference.module_mapper import (
+    PythonModuleOwners,
+    PythonModuleOwnersRequest,
+)
+from pants.backend.python.dependency_inference.rules import import_rules
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import PythonResolveField
+from pants.engine.fs import GlobMatchErrorBehavior, PathGlobs, Paths
+from pants.engine.rules import Get, collect_rules, MultiGet, rule, UnionRule
+from pants.engine.target import (
+    Dependencies,
+    DependenciesRequest,
+    ExplicitlyProvidedDependencies,
+    FieldSet,
+    InferDependenciesRequest,
+    InferredDependencies,
+    InvalidFieldException,
+)
+from pants.source.source_root import SourceRoot, SourceRootRequest
+from pants.util.logging import LogLevel
+
+from stevedore_extensions.target_types import (
+    ResolvedStevedoreEntryPoints,
+    ResolveStevedoreEntryPointsRequest,
+    StevedoreEntryPoints,
+    StevedoreEntryPointsField,
+)
+
+
+# -----------------------------------------------------------------------------------------------
+# `StevedoreExtension` target rules
+# -----------------------------------------------------------------------------------------------
+
+
+@rule(
+    desc="Determining the entry points for a `stevedore_extension` target",
+    level=LogLevel.DEBUG,
+)
+async def resolve_stevedore_entry_points(
+    request: ResolveStevedoreEntryPointsRequest,
+) -> ResolvedStevedoreEntryPoints:
+    # based on: pants.backend.python.target_types_rules.resolve_pex_entry_point
```

Comment:
You can see this here: https://github.com/pantsbuild/pants/blob/2.14.x/src/python/pants/backend/python/target_types_rules.py#L146-L199